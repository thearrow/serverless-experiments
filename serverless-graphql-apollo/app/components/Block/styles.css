.main {
  background: #FFF;
  margin-bottom: 2rem;
}

.padding {
  padding: 1.5rem;
}
