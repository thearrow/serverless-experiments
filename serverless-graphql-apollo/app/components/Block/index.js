import React from 'react';
import classnames from 'classnames';
import styles from './styles.css';

function classes(props) {
  return classnames(styles.main, props.className, {
    [styles.padding]: props.padding,
  });
}

const Block = (props) => (
  <div className={classes(props)}>
    {props.children}
  </div>
);

export default Block;
