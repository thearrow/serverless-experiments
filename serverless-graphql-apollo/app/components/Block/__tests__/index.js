import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Block from '../index';

describe('Block', () => {
  it('renders the content', () => {
    const component = renderer.create(
      <Block>Content</Block>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders the content with the padding className', () => {
    const component = renderer.create(
      <Block padding>Content</Block>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders the content with a custom provided className', () => {
    const component = renderer.create(
      <Block className="test">Content</Block>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
