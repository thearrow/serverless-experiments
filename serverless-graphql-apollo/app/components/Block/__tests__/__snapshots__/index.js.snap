exports[`Block renders the content 1`] = `
<div
  className="main">
  Content
</div>
`;

exports[`Block renders the content with a custom provided className 1`] = `
<div
  className="main test">
  Content
</div>
`;

exports[`Block renders the content with the padding className 1`] = `
<div
  className="main padding">
  Content
</div>
`;
